crates/autopilot/src/domain/settlement/trade/math.rs

```diff
@@ -158,10 +158,16 @@ impl Trade {
     pub fn protocol_fees_in_sell_token(
         &self,
         auction: &settlement::Auction,
-    ) -> Result<Vec<(eth::SellTokenAmount, fee::Policy)>, Error> {
+    ) -> Result<Vec<(eth::TokenAddress, eth::SellTokenAmount, fee::Policy)>, Error> {
```

Comment:
This triple should probably become it's own struct (looks like ExecutedProtocolFee could be used if policy becomes a part of it)

Reply:
refactored